File: backend/app/models/inventory_count.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, DateTime, Text
from ..database import Base


class InventoryCount(Base):
    __tablename__ = "inventory_counts"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    tenant_id = Column(String(50), nullable=False, index=True)
    user_id = Column(String(150), nullable=False)
    user_name = Column(String(150), default="")
    product_id = Column(String(150), nullable=False)
    product_name = Column(String(255), nullable=False)
    sku = Column(String(100), default="")
    warehouse = Column(String(100), default="")
    shelf = Column(String(50), default="")
    row = Column(String(50), default="")
    box = Column(String(50), default="")
    scanned_count = Column(Integer, default=0)
    status = Column(String(20), default="pending")
    session_id = Column(String(150), nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    completed_at = Column(DateTime, nullable=True)

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/InventoryCount.cs
using System.ComponentModel.DataAnnotations.Schema;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("inventory_counts")]
public class InventoryCount : BaseEntity
{
    [Column("tenant_id")]
    public string TenantId { get; set; } = "default";

    [Column("user_id")]
    public string UserId { get; set; } = string.Empty;

    [Column("user_name")]
    public string UserName { get; set; } = string.Empty;

    [Column("product_id")]
    public string ProductId { get; set; } = string.Empty;

    [Column("product_name")]
    public string ProductName { get; set; } = string.Empty;

    [Column("sku")]
    public string Sku { get; set; } = string.Empty;

    [Column("warehouse")]
    public string? Warehouse { get; set; }

    [Column("shelf")]
    public string? Shelf { get; set; }

    [Column("row")]
    public string? Row { get; set; }

    [Column("box")]
    public string? Box { get; set; }

    [Column("scanned_count")]
    public decimal ScannedCount { get; set; }

    [Column("status")]
    public string Status { get; set; } = "pending";

    [Column("session_id")]
    public string? SessionId { get; set; }

    [Column("completed_at")]
    public DateTime? CompletedAt { get; set; }
}
